Netable/Netable/Netable.swift

```diff
@@ -111,7 +107,7 @@ open class Netable {
                     throw NetableError.requestFailed(error)
                 }
 
-                guard let response = response as? HTTPURLResponse else { fatalError("Casting response to HTTPURLResponse failed") }
+                guard let response = response as? HTTPURLResponse else { throw NetableError.codingError("Casting response to HTTPURLResponse failed") }
```

Comment:
I don't think `codingError` is the right thing to send here. `codingError` is documented as a error in encoding the parameters (it maybe should be called `encodingError`).

If we were going to throw an error here, .decodingError would probably be more appropriate.

HOWEVER, although it's very unclear from the documentation when a non-`HTTPURLResponse` response would come back, I think that probably this would only happen if you specified a valid non-HTTP protocol (i.e. `netable.baseURL = "ftp://wtf.com/something.json"`). So assuming we aren't going to add ftp and gopher support (😄) probably checking that the url scheme is `http` or `https` up front and throwing a .malformedURL would be a better approach rather than waiting for the successful response and throwing an unrelated error here. If we were doing that, I think we could safely leave this as a fatal error (since if we suppress non-HTTP schemes, this should only happen due to a bug in the library or Foundation itself, and would never be a recoverable error for the user).

Reply:
Changed back to a fatalError and added an explict check for supported schemes!